contrib/win32/win32compat/misc.c

```diff
@@ -1413,3 +1435,28 @@ get_others_file_permissions(wchar_t * file_name, int isReadOnlyFile)
 		LocalFree(pSD);
 	return ret;
 }
+
+char*
+get_program_data_path()
+{
+	if (ssh_cfg_dir_path[0]) return ssh_cfg_dir_path;
+
+	int return_val = ExpandEnvironmentStringsA("%programData%", ssh_cfg_dir_path, PATH_MAX);
```

Comment:
Use _wdupenv_s and do a utf16_to_8
ExpandEnvironmentStringsA is an overkill here.